openedx/adg/lms/webinars/models.py

```diff
@@ -438,6 +441,68 @@ def get_registered_and_unregistered_team_members(self):
+
+        1. Send invites to all team members when webinar is published for the first time.
+        2. Once a webinar is published then send invites to only new members.
+
+        Returns:
+            set: Contains emails of webinar team members.
+        """
+        old_webinar_data = getattr(self, '_loaded_values', {})
+        team_emails_for_invitations = self.team_emails
+
+        if old_webinar_data and old_webinar_data.get('is_published'):
+            old_panelists = extract_emails_from_string(old_webinar_data.get('panelists'))
+            old_alternative_hosts = extract_emails_from_string(old_webinar_data.get('alternative_hosts'))
+            old_team_members = set(old_panelists).union(set(old_alternative_hosts))
+
+            team_emails_for_invitations = team_emails_for_invitations.difference(old_team_members)
+
+        return team_emails_for_invitations
+
+    def get_invitee_recipients_for_invitation_emails(self, are_webinar_details_updated):
+        """
+        Get invitee emails (invited by email + belonging to a cohort) to which webinar invite emails will be sent.
+
+        1. Send invites to all the invitees when a webinar is published for the first time.
+        2. Once a webinar is published then:
+            i. If webinar details like title, description, start time are updated, Send invites to all those who
+                haven't registered in the webinar.
+            ii. Else send invites to newly added invitees.
+
+        Arguments:
+            are_webinar_details_updated (bool): Are webinar details like title, description, start time updated.
+
+        Returns:
+            set: Contains emails to which invites will be sent.
+        """
+        invited_emails = extract_emails_from_string(self.invites_by_email_address)
+        if self.invited_user_cohort:
+            invited_emails += Webinar.get_user_emails_of_invited_cohort(self.invited_user_cohort)
+
+        old_webinar_data = getattr(self, '_loaded_values', {})
+        if not old_webinar_data or not old_webinar_data.get('is_published'):
+            return set(invited_emails)
+
+        # Send emails to all invitees when webinar details like title, description, start time are updated.
+        user_emails_registered_for_webinar = set(
+            self.registrations.registered_user_emails()  # pylint: disable=no-member
+        )
+        if are_webinar_details_updated:
+            return set(invited_emails).difference(user_emails_registered_for_webinar)
+
+        old_invited_emails = extract_emails_from_string(old_webinar_data.get('invites_by_email_address'))
+        old_invited_cohort = old_webinar_data.get('invited_user_cohort')
+        if old_invited_cohort:
+            old_invited_emails += Webinar.get_user_emails_of_invited_cohort(old_invited_cohort)
+
+        invited_emails = set(invited_emails).difference(set(old_invited_emails))
+
+        return invited_emails.difference(user_emails_registered_for_webinar)
```

Comment:
Let's add some more comments in this methods.
Maybe we can add a comment where each case starts like `#  webinar is published for the first time` and so on and so forth.